src/media/js/site/login.js

```diff
@@ -14,8 +14,39 @@ export class DefaultUnauthorizedHandler extends React.Component {
 };
 
 
+export class DefaultTOSSignatureHandler extends React.Component {
```

Comment:
more complexity in implementation, but cleaner abstraction since TOS doesn't really relate to login. i could be convinced to look the other way though...